script.js

```diff
@@ -1,4 +1,146 @@
+  if (gameStage == PLAYER1) {
+    player1DiceNums = diceNums;
+    console.log(`p1globalnums ` + player1DiceNums);
+    return `Hi ${gameStage}, you rolled ${dice1} for Dice 1 and ${dice2} for Dice 2. <br><br>${selectDiceOrderMessage}`;
+  } else {
+    player2DiceNums = diceNums;
+    console.log(`p2globalnums ` + player2DiceNums);
+    return `Hi ${gameStage}, you rolled ${dice1} for Dice 1 and ${dice2} for Dice 2. <br><br>${selectDiceOrderMessage}`;
+  }
+};
+
+// calculating players' score based on the dice order they selected
+var getPlayersScore = function (input) {
+  if (gameStage == PLAYER1_CHOOSE_DICE_ORDER) {
+    if (input == 1) {
+      player1Score = Number(
+        String(player1DiceNums[0]) + String(player1DiceNums[1])
+      );
+      console.log(`p1 player score 1 ` + player1Score);
+    } else {
+      player1Score = Number(
+        String(player1DiceNums[1]) + String(player1DiceNums[0])
+      );
+      console.log(`P1 player score 2 ` + player1Score);
+    }
+    totalScorePlayer1 = totalScorePlayer1 + player1Score;
+    return `Player 1, your score is ${player1Score}. <br><br> ${totalScoreMessage} ${totalScorePlayer1} <br><br>Click Submit for Player 2 to roll the dices!`;
+  }
+  if (gameStage == PLAYER2_CHOOSE_DICE_ORDER) {
+    if (input == 1) {
+      player2Score = Number(
+        String(player2DiceNums[0]) + String(player2DiceNums[1])
+      );
+      console.log(`P2 player score 1 ` + player2Score);
+    } else {
+      player2Score = Number(
+        String(player2DiceNums[1]) + String(player2DiceNums[0])
+      );
+      console.log(`P2 player score 2 ` + player2Score);
+    }
+    totalScorePlayer2 = totalScorePlayer2 + player2Score;
+    return `Player 2, your score is ${player2Score}. <br><br>${totalScoreMessage} ${totalScorePlayer2} <br><br>Click Submit to see who won the game!`;
+  }
+};
+
+// comparing the players total score
+var compareScore = function () {
+  if (player1Score > player2Score) {
+    return `Congrats Player 1, you won this round. <br><br> Your score: ${player1Score} <br> Player 2 score: ${player2Score}${submitToRestartGameMessage}<br><br> ${determineLeaderNow()}`;
+  }
+  if (player1Score < player2Score) {
+    return `Congrats Player 2, you won this round. <br><br>Your score: ${player2Score} <br> Player 1 score: ${player1Score}${submitToRestartGameMessage}<br><br> ${determineLeaderNow()}`;
+  }
+  if (player1Score == player2Score) {
+    return `Oops its a draw! <br><br>Both of you have the same score :${player1Score}!${submitToRestartGameMessage}<br><br> ${determineLeaderNow()}`;
+  }
+};
+
+// creating leaderboard
+var determineLeaderNow = function () {
+  if (totalScorePlayer1 > totalScorePlayer2) {
+    return `LeaderBoard<br>1. Player 1, Total Score: ${totalScorePlayer1}<br> 2. Player 2, Total Score: ${totalScorePlayer2}`;
+  } else {
+    return `LeaderBoard<br>1. Player 2, Total Score: ${totalScorePlayer2}<br> 2. Player 1, Total Score: ${totalScorePlayer1}`;
+  }
+};
```

Comment:
Nice breaking up of the code to all these smaller functions that do just one thing. (Separation of concerns)